envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/EnvoyListenersFactory.kt

```diff
@@ -127,39 +160,21 @@ class EnvoyListenersFactory(
         group: Group,
         listenersConfig: ListenersConfig,
         globalSnapshot: GlobalSnapshot,
-        transportProtocol: String = "raw_buffer"
+        transportProtocol: TransportProtocol = TransportProtocol.RAW_BUFFER
     ): FilterChain.Builder {
+        val statPrefix = if (transportProtocol == TransportProtocol.RAW_BUFFER) "ingress_http" else "ingress_https"
```

Comment:
I think 
```
val statPrefix = when(transportProtocol) {
  RAW_BUFFER -> "ingress_http"
  TLS -> "ingress_https"
}
```

would be nicer because the mappings are explicit and it will fail it there is unhandled TransportProtocol enum value